src/elm/Auth.elm

```diff
@@ -189,81 +274,182 @@ view isModal shared model =
+                        , onInput EnteredPassphrase
+                        , required True
+                        , autocomplete False
+                        ]
+                        []
+                    , div [ class "input-label pr-1 absolute right-0 text-white font-bold mt-1 text-right" ]
+                        [ text <|
+                            tr
+                                "edit.input_counter"
+                                [ ( "current", passphraseWordsCount )
+                                , ( "max", "12" )
+                                ]
+                        ]
+                    ]
+                , ul [ class "form-error-on-dark-bg absolute" ] errors
+                ]
+            , div [ class "sf-footer" ]
+                [ if not isModal then
+                    p [ class "text-white text-body text-center mt-16 mb-6 block" ]
+                        [ text_ "auth.login.register"
+                        , a [ Route.href (Route.Register Nothing Nothing), class "text-orange-300 underline" ] [ text_ "auth.login.registerLink" ]
+                        ]
+
+                  else
+                    text ""
+                , button
+                    [ class buttonClass
+                    , onClick ClickedViewLoginPinStep
+                    ]
+                    [ text_ "dashboard.next" ]
+                ]
             ]
 
-      else
-        text ""
+        viewCreatePin =
+            let
+                trPrefix s =
+                    "auth.pin.instruction." ++ s
+            in
+            [ div [ class "sf-content" ]
+                [ illustration "login_pin.svg"
+                , p [ class pClass ]
+                    [ text_ (trPrefix "nowCreate")
+                    , text " "
+                    , strong [] [ text_ (trPrefix "sixDigitPin") ]
+                    , text ". "
+                    , text_ (trPrefix "thePin")
+                    , text " "
+                    , strong [] [ text_ (trPrefix "notPassword") ]
+                    , text " "
+                    , text_ (trPrefix "eachLogin")
+                    ]
+                , viewPin model shared
+                , viewPinConfirmation model shared
+                ]
+            , div [ class "sf-footer" ]
+                [ button
+                    [ class buttonClass
+                    , class "mt-10"
+                    , onClick (SubmittedLoginPrivateKey model.form)
+                    ]
+                    [ text_ "auth.login.submit" ]
+                ]
+            ]
+    in
+    [ div
+        [ class "px-4 md:max-w-sm md:mx-auto md:pt-20 md:px-0"
+        , class "sf-wrapper w-full"
+        ]
+        ([ viewAuthError shared model.loginError ]
+            ++ (case loginStep of
+                    LoginStepPassphrase ->
+                        viewPassphrase
+
+                    LoginStepPIN ->
+                        viewCreatePin
+               )
+        )
```

Comment:
why not this:

```elm
[ div [ . . . ]
        [ viewAuthError shared model.loginError
        , case loginSetup of
             . . .
]

Reply:
In this case, we would have different types in a list:

- `viewAuthError ...` is `Html Msg`,
- but `case loginSetup of...` is `List (Html Msg)`.

Anyway, this code looks complicated, I made it more readable.